src/App.tsx

```diff
@@ -1,19 +1,66 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { debounce } from 'lodash';
+
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [hasClick, setHasClick] = useState(false);
+  const [person, setPerson] = useState<Person | null>(null);
+  const [focus, setFocus] = useState(false);
+  const [query, setQuery] = useState('');
+  const [aplliedQuery, setAplliedQuery] = useState('');
+
+  const queryDetention = useCallback(
+    debounce(setAplliedQuery, 1000),
+    [],
+  );
+
+  let preparedPeopleData = [...peopleFromServer];
+
+  if (aplliedQuery) {
+    preparedPeopleData = preparedPeopleData.filter(man => {
+      const name = man.name.toUpperCase();
+
+      return name.includes(aplliedQuery.toUpperCase());
+    });
+  }
+
+  useEffect(() => setFocus(false), [person, hasClick]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.currentTarget.value);
+    queryDetention(event.currentTarget.value);
+  };
+
+  const handleClick = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    currentPerson: Person,
+  ) => {
+    event.preventDefault();
+    setQuery(currentPerson.name);
+    setHasClick(!hasClick);
+    setPerson(currentPerson);
+  };
 
   return (
     <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+      {preparedPeopleData.length && person && query ? (
+        <h1 className="title">
+          {`${person.name} (${person.born} - ${person.died})`}
+        </h1>
+      ) : (
+        <h1 className="title">No matching suggestions</h1>
+      )}
 
-      <div className="dropdown is-active">
+      <div className={`dropdown ${focus && 'is-active'}`}>
```

Comment:
suggest to use [classnames](https://www.npmjs.com/package/classnames) library for dynamic build classnames. it is avoid such useless classes  as false
```
import cn from 'classnames'
```
```suggestion
      <div className={cn('dropdown', { 'is-active' :  focus })>
```
![Screenshot from 2023-08-16 13-03-55](https://github.com/mate-academy/react_autocomplete/assets/33029864/51cd3306-e208-4b07-b8d0-280069056532)